python/cudf/cudf/core/index.py

```diff
@@ -1607,25 +1610,28 @@ def __repr__(self):
         )
 
     def __len__(self):
-        return max(0, self._stop - self._start)
+        return len(range(self._start, self._stop, self._step))
 
     def __getitem__(self, index):
         if isinstance(index, slice):
-            start, stop, step = index.indices(len(self))
-            sln = (stop - start) // step
-            sln = max(0, sln)
-            start += self._start
-            stop += self._start
-            if sln == 0:
-                return RangeIndex(0, stop=None, name=self.name)
-            elif step == 1:
-                return RangeIndex(start, stop=stop, name=self.name)
-            else:
-                return index_from_range(start, stop, step)
+            # sl_start = 0 if index.start is None else index.start
+            # sl_stop = len(self) if index.stop is None else index.stop
+            # sl_step = 1 if index.step is None else index.step
+
+            # if len(self) == 0:
```

Comment:
Stale code. BTW, this is a really good read regarding this: https://nedbatchelder.com/text/deleting-code.html

Reply:
Just read the "deleting-code" article, I'm pretty sure this belongs to the `Accidental droppings` category. Will attempt to use a distinctive mark in the future.